Backend/app/kafka-consumers/src/DriverTrackingHealthCheck/Service/Runner.hs

```diff
@@ -0,0 +1,128 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+{-# OPTIONS_GHC -Wno-unused-do-bind #-}
+
+module DriverTrackingHealthCheck.Service.Runner where
+
+import Consumer.LocationUpdate.Types (DriverIdTokenKey)
+import qualified Data.Aeson as A
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as BSL
+import Data.List.NonEmpty (nonEmpty)
+import Data.Time.Clock.POSIX hiding (getCurrentTime)
+import Domain.Types.DriverInformation (DriverMode (..))
+import qualified "dynamic-offer-driver-app" Domain.Types.Person as SP
+import qualified DriverTrackingHealthCheck.API as HC
+import Environment (Flow, HealthCheckAppCfg)
+import Kernel.External.Encryption (decrypt)
+import qualified Kernel.External.Notification.FCM.Types as FCM
+import qualified Kernel.External.SMS.MyValueFirst.Flow as SF
+import Kernel.Prelude
+import Kernel.Storage.Hedis (lPush, rPop)
+import qualified Kernel.Storage.Hedis as Redis
+import Kernel.Types.Common
+import Kernel.Types.Error (PersonError (PersonFieldNotPresent, PersonNotFound))
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import Kernel.Utils.Service
+import qualified Storage.Queries.DriverInformation as DrInfo
+import qualified "dynamic-offer-driver-app" Storage.Queries.Person as SQP
+import "dynamic-offer-driver-app" Tools.Notifications
+
+driverTrackingHealthcheckService :: HealthCheckAppCfg -> Flow ()
+driverTrackingHealthcheckService heathCheckConfig = withLogTag "driverTrackingHealthcheckService" do
+  driverLastLocationUpdateCheckService heathCheckConfig
+  driverDevicePingService heathCheckConfig
+
+-- driverMakingInactiveService heathCheckConfig --not need to run inactive service
+
+driverLastLocationUpdateCheckService :: HealthCheckAppCfg -> Flow ()
+driverLastLocationUpdateCheckService healthCheckAppCfg = startService "driverLastLocationUpdateCheckService" $ withRandomId do
+  let locationDelay = healthCheckAppCfg.driverAllowedDelayForLocationUpdateInSec
+      serviceInterval = healthCheckAppCfg.driverLocationHealthCheckIntervalInSec
+  withLock "driver-tracking-healthcheck" $ measuringDurationToLog INFO "driverLastLocationUpdateCheckService" do
+    now <- getCurrentTime
+    HC.iAmAlive
+    drivers <- getAllDrivers locationDelay now
+    case nonEmpty drivers of
+      Just allDrivers -> do
+        lPush redisKey allDrivers
```

Comment:
lets merge these two functions instead of lpush and rpop thing....fetch and send FCM here itself with the loop running at x interval.